src/parser/warlock/affliction/modules/features/Darkglare.js

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+import Enemies from 'parser/core/modules/Enemies';
+import { encodeTargetString } from 'parser/core/modules/EnemyInstances';
+
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatThousands } from 'common/format';
+
+import StatisticsListBox from 'interface/others/StatisticsListBox';
+import StatisticListBoxItem from 'interface/others/StatisticListBoxItem';
+
+import { UNSTABLE_AFFLICTION_DEBUFF_IDS } from '../../constants';
+
+const BONUS_DURATION = 8000;
+const DOTS_AFFECTED_BY_CREEPING_DEATH = [
+  SPELLS.AGONY.id,
+  SPELLS.CORRUPTION_DEBUFF.id,
+  SPELLS.SIPHON_LIFE_TALENT.id,
+  ...UNSTABLE_AFFLICTION_DEBUFF_IDS,
+];
+const DOT_DEBUFF_IDS = [
+  SPELLS.PHANTOM_SINGULARITY_TALENT.id,
+  ...DOTS_AFFECTED_BY_CREEPING_DEATH,
+];
+const debug = false;
+
+class Darkglare extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  _dotDurations = {
+    [SPELLS.AGONY.id]: 18000,
+    [SPELLS.CORRUPTION_DEBUFF.id]: 14000,
+    [SPELLS.SIPHON_LIFE_TALENT.id]: 15000,
+    // UA IDs added in constructor
+    [SPELLS.PHANTOM_SINGULARITY_TALENT.id]: 16000,
+  };
+  _hasAC = false;
+
+  bonusDotDamage = 0;
+  darkglareDamage = 0;
+  casts = [
+    /*
+      {
+        timestamp: number
+        [encoded target string]: {
+          targetName: name,
+          dots: [dot IDs],
+        }
+      },
+     */
+  ];
+  dots = {
+    /*[encoded target string]: {
+        targetName: name
+        dots: {
+          [dot ID]: {
+            cast: timestamp
+            expectedEnd: timestamp,
+            extendStart: timestamp | null,
+            extendExpectedEnd: timestamp | null
+          },
+        }
+      },
+    */
+  };
+
+  constructor(...args) {
+    super(...args);
+    // if player has Absolute Corruption, disregard the Corruption duration (it's permanent debuff then)
+    this._hasAC = this.selectedCombatant.hasTalent(SPELLS.ABSOLUTE_CORRUPTION_TALENT.id);
+    if (this._hasAC) {
+      delete this._dotDurations[SPELLS.CORRUPTION_DEBUFF.id];
+    }
+    UNSTABLE_AFFLICTION_DEBUFF_IDS.forEach(id => {
+      this._dotDurations[id] = 8000;
```

Comment:
This is the affliction duration if I read this code correct and not the dark glare's bonus duration.
Would make it cleaner if you put that in a const for better readability.